// nexmo/lib/Number.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _index = require("./index");

var _index2 = _interopRequireDefault(_index);

var _Pricing = require("./Pricing");

var _Pricing2 = _interopRequireDefault(_Pricing);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Number = function () {
  /**
   * @param {Credentials} credentials
   *    credentials to be used when interacting with the API.
   * @param {Object} options
   *    Addition Number options.
   */
  function Number(credentials) {
    var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    _classCallCheck(this, Number);

    this.creds = credentials;
    this.options = options;

    this._pricing = new _Pricing2.default(credentials, options);

    // Used to facilitate testing of the call to the underlying object
    this._nexmo = this.options.nexmoOverride || _index2.default;

    this._nexmo.initialize(this.creds.apiKey, this.creds.apiSecret, this.options);
  }

  /**
   * TODO: remove with next major release
   */


  _createClass(Number, [{
    key: "getPricing",
    value: function getPricing() {
      this._pricing.get.apply(this, arguments);
    }

    /**
     * TODO: remove with next major release
     */

  }, {
    key: "getPhonePricing",
    value: function getPhonePricing() {
      this._pricing.getPhone.apply(this, arguments);
    }

    /**
     * TODO: document
     */

  }, {
    key: "get",
    value: function get() {
      this._nexmo.getNumbers.apply(this._nexmo, arguments);
    }

    /**
     * TODO: document
     */

  }, {
    key: "search",
    value: function search() {
      this._nexmo.searchNumbers.apply(this._nexmo, arguments);
    }

    /**
     * TODO: document
     */

  }, {
    key: "buy",
    value: function buy() {
      this._nexmo.buyNumber.apply(this._nexmo, arguments);
    }

    /**
     * TODO: document
     */

  }, {
    key: "cancel",
    value: function cancel() {
      this._nexmo.cancelNumber.apply(this._nexmo, arguments);
    }

    /**
     * TODO: document
     */

  }, {
    key: "update",
    value: function update() {
      this._nexmo.updateNumber.apply(this._nexmo, arguments);
    }
  }]);

  return Number;
}();

exports.default = Number;
module.exports = exports["default"];
