// nexmo/lib/Pricing.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _Utils = require("./Utils");

var _Utils2 = _interopRequireDefault(_Utils);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var Pricing = function () {
  _createClass(Pricing, null, [{
    key: "PATH",
    get: function get() {
      return "/account/{endpoint}/outbound/{type}";
    }
  }]);

  function Pricing(credentials, options) {
    _classCallCheck(this, Pricing);

    this.creds = credentials;
    this.options = options;
  }

  _createClass(Pricing, [{
    key: "get",
    value: function get(type, country, callback) {
      return this.options.rest.get(Pricing.PATH.replace("{endpoint}", "get-pricing").replace("{type}", type), { country: country }, callback);
    }
  }, {
    key: "getFull",
    value: function getFull(type, callback) {
      return this.options.rest.get(Pricing.PATH.replace("{endpoint}", "get-full-pricing").replace("{type}", type), callback);
    }
  }, {
    key: "getPrefix",
    value: function getPrefix(type, prefix, callback) {
      return this.options.rest.get(Pricing.PATH.replace("{endpoint}", "get-prefix-pricing").replace("{type}", type), { prefix: prefix }, callback);
    }
  }, {
    key: "getPhone",
    value: function getPhone(type, phone, callback) {
      return this.options.rest.get(Pricing.PATH.replace("{endpoint}", "get-phone-pricing").replace("{type}", type), { phone: phone }, callback);
    }
  }]);

  return Pricing;
}();

exports.default = Pricing;
module.exports = exports["default"];
